adapters/buenos-aires.js

```diff
@@ -0,0 +1,192 @@
+        }
+        const link = [
+          source,
+          pollutantNum,
+          '&estacion=',
+          stationNum,
+          '&fecha_dia=',
+          day,
+          '&fecha_mes=',
+          month,
+          '&fecha_anio=',
+          year,
+          '&menu_id=34234&buscar=Buscar'
+        ].join('');
+        const linkRequest = (callback) => {
+          request.get({
+            url: link
+          }, (err, res, body) => {
+            if (err || res.statusCode !== 200) {
+              return callback(new Error('was not able to retrieve' + pollutant + ' data for ' + station));
+            }
+            callback(null, body);
+          });
+        };
+        linkRequests.push(linkRequest);
+      };
+      makeLinks(station, pollutant);
+    });
+  });
+  return linkRequests;
+};
+
+// the below three set*** functions set the measurement attribute per the passed pollutant or station.
+const setCoordinates = (estacion) => {
+  switch (estacion) {
+    case 'CENTENARIO':
+      return { latitude: -34.60638, longitude: -58.43194 };
+    case 'CORDOBA':
+      return { latitude: -34.60441667, longitude: -58.39165 };
+    case 'LA BOCA':
+      return { latitude: -34.62527, longitude: -58.36555 };
+    default:
+      break;
+  }
+};
+const setUnits = (contaminante) => {
+  switch (contaminante) {
+    case 'CO':
+      return 'ppm';
+    case 'NO2':
+      return 'ppb';
+    case 'PM10':
+      return 'µg/m3';
+    default:
+      break;
+  }
+};
+
+const setPeriod = (estacion) => {
+  switch (estacion) {
+    case 'CO':
+      return {unit: 'hours', value: 8};
+    case 'NO2':
+      return {unit: 'hours', value: 1};
+    case 'PM10':
+      return {unit: 'hours', value: 24};
+    default:
+      break;
+  }
+};
+
+// returns list of measurements generated from requests.
+const formatData = (htmlList) => {
+  let measurements = htmlList.map((html) => {
+    // data is in imgs' title tags like [pollutantVal - hr]
+    // aqData is list of lists with said data
+    let aqData = html.match(/<img[^>]+>/g).filter((img) => {
+      return img.match(/hs/);
+    });
+    aqData = aqData.map((img) => {
+      return img
+        .split('title=')[1]
+        .split("'")[1]
+        .replace(' hs.', '')
+        .split(' - ');
+    })[aqData.length - 1];
+    // take hr from aqData and make into proper date
+    aqData[1] = moment().startOf('day').add(aqData[1], 'h');
+    aqData[1] = moment.tz(
+      aqData[1],
+      'DD/MM/YYYY HH:mm:ss',
+      'America/Argentina/Buenos_Aires'
+    );
+    // get contaminante, pollutant, and estacion, station. use them in above set** functions
+    // when populating measurement
+    const $ = cheerio.load(html);
+    const contaminante = $('#contaminante').html().split('selected>')[1].split('<')[0];
+    const estacion = $('#estacion').html().split('selected>')[1].split('<')[0];
+    const measurement = {};
+    measurement['parameter'] = contaminante === 'PM10' ? 'pm10' : contaminante;
```

Comment:
This is totally personal preference, so feel free to ignore, but I try to write these like

```js
const measurements = {
  parameter: contaminante === 'PM10' ? 'pm10' : contaminante,
  date: {
    utc: aqData[1].toDate(),
    local: aqData[1].format()
  }
};
```
I feel it reads better, but like I said, totally personal preference.